File: modul_praktikum_python/contoh_code/pascal/matriks_3x3.pas
uses crt; 
type  
  tmatrik=array[1..3,1..3]of integer; 

procedure isi(var matrik:tmatrik); 
var 
 i,j:integer; 
begin 
 for i:=1 to 3 do   
  for j:=1 to 3 do 
    begin   
     write(‘A[‘,i,’,’,j,’] = ‘);readln(matrik[i,j]); 
    end; 
end; 

procedure cetak(matrik:tmatrik); 
var 
 i,j:integer; 
begin  
 for i:=1 to 3 do 
 begin   
  for j:=1 to 3 do   
     write(matrik[i,j]:3); 
  writeln; 
 end; 
end; 

var 
 matrik:tmatrik; 
begin 
  writeln(‘Mengisi elemen matrik A’); 
  isi(matrik); 
  writeln(‘Mencetak elemen matrik A’); 
  cetak(matrik); 
end.

File: modul_praktikum_python/contoh_code/python/matriks_3x3.py
"""Matriks 3x3: isi dan cetak."""

# mat[i][j] untuk i,j in 1..3 - baris/kolom 0 tidak dipakai.


def isi(matrik: list[list[int]]) -> None:
    for i in range(1, 4):
        for j in range(1, 4):
            matrik[i][j] = int(input(f"A[{i},{j}] = "))


def cetak(matrik: list[list[int]]) -> None:
    for i in range(1, 4):
        for j in range(1, 4):
            print(f"{matrik[i][j]:3d}", end="")
        print()


def main() -> None:
    matrik = [[0] * 4 for _ in range(4)]
    print("Mengisi elemen matrik A")
    isi(matrik)
    print("Mencetak elemen matrik A")
    cetak(matrik)


if __name__ == "__main__":
    main()
